src/aldente.cpp

```diff
@@ -1,551 +1,99 @@
-	for (int i = 0; i < toAdd.size(); i++) {
-		vector<Tile*> currRow = toAdd[i];
-		for (int j = 0; j < currRow.size(); j++) {
-			SceneModel *currTile = new SceneModel(scene);
-			currTile->add_mesh(currRow[j]->getMesh());
-			/*if (currRow[j]->getRigid() == NULL) {
-				SceneTransform *tileTranslate = new SceneTransform(scene,
-				  glm::translate(glm::mat4(1.f), glm::vec3(currRow[j]->getX(), 0, currRow[j]->getZ())));
-				tileTranslate->add_child(currTile);
-				scene->root->add_child(tileTranslate);
-			}
-			else {*/
-				scene->root->add_child(currTile);
-			//}
-
-			if (currRow[j]->getRigid() != NULL) {
-				dynamicsWorld->addRigidBody(currRow[j]->getRigid());
-				rigidBodies.push_back(currRow[j]->getRigid());
-			}
-
-		}
-	}
-
-	/*
-    // Plane
-    Geometry *plane_geo = GeometryGenerator::generate_plane(1.f, 0);
-    Material plane_mat;
-    plane_mat.diffuse = plane_mat.ambient = color::indian_red;
-    Mesh plane_mesh = { plane_geo, plane_mat, ShaderManager::get_default(), glm::mat4(1.f) };
-        SceneModel *plane_model = new SceneModel(scene);
-        plane_model->add_mesh(plane_mesh);
-        SceneTransform *plane_scale = new SceneTransform(scene, glm::scale(glm::mat4(1.f), glm::vec3(50.f, 1.0f, 50.f)));
-        SceneTransform *plane_translate = new SceneTransform(scene, glm::translate(glm::mat4(1.f), glm::vec3(0.0f, -0.5f * PLAYER_HEIGHT, 0.0f)));
-        plane_scale->add_child(plane_model);
-        plane_translate->add_child(plane_scale);
-        scene->root->add_child(plane_translate);
-	*/
 }
 
 void Aldente::go()
 {
-    // Get window width and height from config
-    int width, height;
-    Config::config->get_value(Config::str_screen_width, width);
-    Config::config->get_value(Config::str_screen_height, height);
-    std::string game_name;
-    Config::config->get_value(Config::str_game_name, game_name);
-
-    window = Window::create_window(width, height, game_name.c_str());
-    //glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_HIDDEN); // Don't show cursor
-    setup_callbacks();
-    setup_opengl();
-
-    setup_shaders();
-
-	// Initialize Bullet. This strictly follows http://bulletphysics.org/mediawiki-1.5.8/index.php/Hello_World,
-	// even though we won't use most of this stuff.
-
-	// Build the broadphase
-	broadphase = new btDbvtBroadphase();
-
-	// Set up the collision configuration and dispatcher
-	collisionConfiguration = new btDefaultCollisionConfiguration();
-	dispatcher = new btCollisionDispatcher(collisionConfiguration);
-
-	// The actual physics solver
-	solver = new btSequentialImpulseConstraintSolver;
-
-	// The world.
-	dynamicsWorld = new btDiscreteDynamicsWorld(dispatcher, broadphase, solver, collisionConfiguration);
-	dynamicsWorld->setGravity(btVector3(0, -9.81f, 0));
-
-	// Seed PRNG.
-    Util::seed(0);
+    window = Window::window->create_window();
+	Setup::setup_callbacks();
+    Setup::setup_opengl();
+    Setup::setup_shaders();
+	Physics::physics->setup_bullet();
```

Comment:
Not a huge fan of the "static pointer" design philosophy. Can we perhaps instantiate these modules on the stack and inject dependencies by reference? For example, if a `Physics p` needs an `AssetLoader a`, construct it as `Phsyics p(a)`.